### PR TITLE
Structures to string

### PR DESCRIPTION
**Pull request by stefanak-michal**

### CODE REVIEW

**Comment by stefanak-michal** [comment]
https://7687.org/packstream/packstream-specification-1.html

https://neo4j.com/docs/cypher-manual/current/syntax/temporal/

https://neo4j.com/docs/cypher-manual/current/syntax/spatial/

File: src/structures/DateTimeZoneId.php [hunk 1]
```diff
@@ -66,11 +66,20 @@ public function nanoseconds(): int
...
     }
 
+    public function __toString(): string
+    {
+        $dt = \DateTime::createFromFormat('U', $this->seconds, new \DateTimeZone('UTC'));
+        $fraction = new \DateInterval('PT0S');
+        $fraction->f = $this->nanoseconds / 1000000000;
+        $dt->add($fraction);
+        $dt->setTimezone(new \DateTimeZone($this->tz_id));
+        return $dt->format('Y-m-d\TH:i:s.uP');
```
Comment (transistive) [inline src/structures/DateTimeZoneId.php:83]: Looks good! I would return the date time like this:

```php
$dt->format(DATE_ISO8601)
```

This way we are sure the format will be accepted in all major languages and frameworks

File: src/structures/LocalDateTime.php [hunk 2]
```diff
@@ -53,4 +53,13 @@ public function nanoseconds(): int
         return $this->nanoseconds;
     }
 
+    public function __toString(): string
+    {
+        $dt = \DateTime::createFromFormat('U', $this->seconds, new \DateTimeZone('UTC'));
+        $fraction = new \DateInterval('PT0S');
+        $fraction->f = $this->nanoseconds / 1000000000;
+        $dt->add($fraction);
+        return $dt->format('Y-m-d\TH:i:s.u');
```
Comment (transistive) [inline src/structures/LocalDateTime.php:62]: Same as before, I would use something like this: `$dt->format(DATE_ISO8601)`

**Review by transistive — APPROVED** [review]
Very nice and fast work! I like the fact that you made the Interface and implemented it on all objects :)

I made a small comment about the date format but it looks perfect otherwise! The comment is just a suggestion but not necessary. Thank you :+1:

File: src/structures/DateTimeZoneId.php (hunk 1 above)
Comment (stefanak-michal) [reply src/structures/DateTimeZoneId.php:83]: I was comparing with response from http api and this was format there.

Comment (transistive) [reply src/structures/DateTimeZoneId.php:83]: In that case, it's perfect!

**Comment by stefanak-michal** [comment]
@Mulkave would you like to wait until you finish OGM and try it with these changes or we can merge it?

**Review by Mulkave — APPROVED** [review]

**Comment by Mulkave** [comment]
@stefanak-michal good stuff! I'd love to test it as soon as possible, no need to wait for OGM to finish.

**Comment by stefanak-michal** [comment]
@Mulkave yeah I'll update one of my project too where it will be very usefull.

**Pull request merged**